examples/chef/common/chef-resource-monitoring-delegates.cpp

```diff
@@ -0,0 +1,156 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app-common/zap-generated/ids/Attributes.h>
+#include <app-common/zap-generated/ids/Clusters.h>
+#include <app/clusters/resource-monitoring-server/resource-monitoring-cluster-objects.h>
+#include <app/clusters/resource-monitoring-server/resource-monitoring-server.h>
+#include <chef-resource-monitoring-delegates.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::ResourceMonitoring;
+using namespace chip::app::Clusters::ActivatedCarbonFilterMonitoring;
+using namespace chip::app::Clusters::HepaFilterMonitoring;
+using chip::Protocols::InteractionModel::Status;
+
+const chip::BitMask<ResourceMonitoring::Feature>
+    gHepaFilterFeatureMap(static_cast<uint32_t>(ResourceMonitoring::Feature::kCondition) |
+                          static_cast<uint32_t>(ResourceMonitoring::Feature::kWarning) |
+                          static_cast<uint32_t>(ResourceMonitoring::Feature::kReplacementProductList));
+const chip::BitMask<ResourceMonitoring::Feature>
+    gActivatedCarbonFeatureMap(static_cast<uint32_t>(ResourceMonitoring::Feature::kCondition) |
+                               static_cast<uint32_t>(ResourceMonitoring::Feature::kWarning) |
+                               static_cast<uint32_t>(ResourceMonitoring::Feature::kReplacementProductList));
+
+static std::unique_ptr<ActivatedCarbonFilterMonitoringDelegate> gActivatedCarbonFilterDelegate = nullptr;
+static std::unique_ptr<ResourceMonitoring::Instance> gActivatedCarbonFilterInstance            = nullptr;
+
+static std::unique_ptr<HepaFilterMonitoringDelegate> gHepaFilterDelegate = nullptr;
+static std::unique_ptr<ResourceMonitoring::Instance> gHepaFilterInstance = nullptr;
```

Comment:
you do not need to initialize unique_ptr with null

Reply:
Fixed done.

Reply:
fixed done